src/main/java/net/coderbot/iris/mixin/vertices/MixinBufferBuilder.java

```diff
@@ -135,14 +152,28 @@ public abstract class MixinBufferBuilder implements BufferVertexConsumer, BlockS
 			this.nextElement();
 		}
 
-		this.putShort(0, currentBlock);
-		this.putShort(2, currentRenderType);
-		this.nextElement();
+		if (iris$isTerrain) {
+			// ENTITY_ELEMENT
+			this.putShort(0, currentBlock);
+			this.putShort(2, currentRenderType);
+			this.nextElement();
+		}
+		// MID_TEXTURE_ELEMENT
 		this.putFloat(0, 0);
 		this.putFloat(4, 0);
 		this.nextElement();
+		// TANGENT_ELEMENT
 		this.putInt(0, 0);
 		this.nextElement();
+		if (iris$isTerrain) {
+			// MID_BLOCK_ELEMENT
+			int posIndex = this.nextElementByte - format.getVertexSize() + 4;
+			float x = buffer.getFloat(posIndex);
+			float y = buffer.getFloat(posIndex + 4);
+			float z = buffer.getFloat(posIndex + 8);
+			this.putInt(0, ExtendedDataHelper.computeMidBlock(x, y, z, currentLocalPosX, currentLocalPosY, currentLocalPosZ));
+			this.nextElement();
+		}
```

Comment:
This code seems to silently assume that MID_BLOCK_ELEMENT is the last element. In addition, would we want to handle text (which uses the TERRAIN vertex format) differently?